public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx

```diff
@@ -208,9 +209,37 @@ export class PanelEditor extends SceneObjectBase<PanelEditorState> {
     this.setState({ showLibraryPanelSaveModal: true });
   };
 
-  public onConfirmSaveLibraryPanel = () => {
-    this.state.vizManager.commitChanges();
-    locationService.partial({ editPanel: null });
+  public onConfirmSaveLibraryPanel = async () => {
+    const sourcePanel = this.state.vizManager.state.sourcePanel.resolve();
+    if (!(sourcePanel.parent instanceof LibraryVizPanel)) {
+      throw new Error('Panel is not a library panel');
+    }
+
+    const gridItem = sourcePanel.parent.parent;
+    if (!(gridItem instanceof SceneGridItem)) {
+      throw new Error('Library panel is not child of SceneGridItem');
+    }
```

Comment:
Needs a sync with changes introduced in https://github.com/grafana/grafana/pull/84710